packages/react/runtime/__test__/list.test.jsx

```diff
@@ -2169,6 +2235,144 @@ describe('list-item JSXSpread', () => {
   });
 });
 
+describe('list-item with platform info attributes', () => {
+  const s1 = __SNAPSHOT__(
+    <view>
+      <text>111</text>
+      <list id='list'>{HOLE}</list>
+    </view>,
+  );
+
+  const s3 = __SNAPSHOT__(
+    <list-item
+      item-key={HOLE}
+      reuse-identifier={HOLE}
+      full-span={HOLE}
+      sticky-top={HOLE}
+      sticky-bottom={HOLE}
+      estimated-height={HOLE}
+      estimated-height-px={HOLE}
+      estimated-main-axis-size-px={HOLE}
+      recyclable={HOLE}
+    >
+      <text>World</text>
+    </list-item>,
+  );
+
+  it('basic list-item with platform info attributes', () => {
+    const b = new SnapshotInstance(s1);
+    b.ensureElements();
+    const root = b.__elements[0];
+
+    {
+      const d0 = new SnapshotInstance(s3);
+      const d1 = new SnapshotInstance(s3);
+      const d2 = new SnapshotInstance(s3);
+      b.insertBefore(d0);
+      b.insertBefore(d1);
+      b.insertBefore(d2);
+
+      d0.setAttribute(0, {
+        'item-key': 'list-item-0',
+        'reuse-identifier': 'A',
+        'full-span': true,
+        'sticky-top': true,
+        'sticky-bottom': false,
+        'estimated-height': 100,
+        'estimated-height-px': 100,
+        'estimated-main-axis-size-px': 100,
+        'recyclable': false,
+      });
+      d1.setAttribute(0, {
+        'item-key': 'list-item-1',
+        'reuse-identifier': 'A',
+        'full-span': false,
+        'sticky-top': false,
+        'sticky-bottom': false,
+        'estimated-height': 100,
+        'estimated-height-px': 100,
+        'estimated-main-axis-size-px': 100,
+        'recyclable': true,
+      });
+      d2.setAttribute(0, {
+        'item-key': 'list-item-2',
+        'reuse-identifier': 'A',
+        'full-span': true,
+        'sticky-top': false,
+        'sticky-bottom': true,
+        'estimated-height': 100,
+        'estimated-height-px': 100,
+        'estimated-main-axis-size-px': 100,
+        'recyclable': false,
+      });
+    }
+
+    __pendingListUpdates.flush();
+
+    expect(root).toMatchInlineSnapshot(`
```

Comment:
Could we call componentAtIndex on the list, so we can assert that "recyclable" (and other virtual attributes) will not be set on list-item by __SetAttribute